JetNews/app/src/main/java/com/example/jetnews/data/posts/impl/BlockingFakePostsRepository.kt

```diff
@@ -37,11 +43,30 @@ class BlockingFakePostsRepository(private val context: Context) : PostsRepositor
         }
     }
 
-    override fun getPost(postId: String, callback: (Result<Post?>) -> Unit) {
-        callback(Result.Success(postsWithResources.find { it.id == postId }))
+    private val favorites = MutableStateFlow<Set<String>>(setOf())
+
+    override suspend fun getPost(postId: String): Result<Post> {
+        return withContext(Dispatchers.IO) {
+            val post = postsWithResources.find { it.id == postId }
+            if (post == null) {
+                Result.Error(IllegalArgumentException("Unable to find post"))
+            } else {
+                Result.Success(post)
+            }
+        }
+    }
+
+    override suspend fun getPosts(): Result<List<Post>> {
+        return Result.Success(postsWithResources)
     }
 
-    override fun getPosts(callback: (Result<List<Post>>) -> Unit) {
-        callback(Result.Success(postsWithResources))
+    override fun getFavorites(): Flow<Set<String>> = favorites
+
+    override suspend fun toggleFavorite(postId: String) {
+        val set = favorites.value.toMutableSet()
+        if (!set.add(postId)) {
```

Comment:
Let's make the `MutableSet<E>.addOrRemove` ext function not private in the other file, move it to a helper class, and reuse it here

Reply:
Seems reasonable. Hesitate to extend something like Map, but this is pretty generic.